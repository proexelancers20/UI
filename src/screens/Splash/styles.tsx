import { StyleSheet } from 'react-native';
import colors from "../../helper/colors";
import { FontFamily } from "../../helper/helper";

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.white,
        alignItems: 'center',
        justifyContent: 'center',
      },
      wrappText: {
        fontSize: 85,
        fontFamily: FontFamily.bold,
      },
})

export default styles;