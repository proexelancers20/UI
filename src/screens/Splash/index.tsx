import React, {useEffect} from 'react';
import {StatusBar, StyleSheet, Text, View} from 'react-native';
import {SplashIcon} from '../../assets/images';
import {scale, verticalScale} from 'react-native-size-matters';
import {CommonString} from '../../helper/helper';
import styles from './styles';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {NavigationProps} from '../../navigation';

type SplashScreenProps = NativeStackScreenProps<
  NavigationProps,
  'SplashScreen'
>;

const SplashScreen: React.FC<SplashScreenProps> = ({navigation}) => {
  useEffect(() => {
    setTimeout(() => {
      navigation.navigate('OnBoardScreen');
    }, 1500);
  });

  return (
    <View style={styles.container}>
      <StatusBar hidden={true} />
      {/* <SplashIcon width={scale(275)} height={verticalScale(60)} /> */}
      <Text testID="SplashWrapp" style={styles.wrappText}>
        {CommonString.SplashTitle}
      </Text>
    </View>
  );
};

export default SplashScreen;
