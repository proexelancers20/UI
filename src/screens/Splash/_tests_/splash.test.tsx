import React from 'react';
import Splash from '..';
import {render} from '@testing-library/react-native';

jest.mock('react-native-size-matters', () => ({
  scale: jest.fn(),
  verticalScale: jest.fn(),
}));
jest.mock('@react-navigation/native-stack', () => ({
  createNativeStackNavigator: () => ({
    Navigator: 'mocked-Navigator',
    Screen: 'mocked-Screen',
  }),
}));
jest.mock('@react-navigation/native', () => ({
  NavigationContainer: jest.fn(),
}));
jest.mock('react-native-svg');
jest.mock('react-native-svg-transformer');
jest.mock('react-native-swiper', () => {
  const React = require('react');
  const Swiper = (props: any) => {
    // Mock implementation of Swiper
    return React.createElement('View', { 'data-testid': 'mock-swiper' }, 'Mock Swiper Component');
  };
  return Swiper;
});

const component = (
  <Splash
    navigation={{ navigate: jest.fn(), goBack: jest.fn(), canGoBack: jest.fn() } as any}
    route={undefined as any}
  />
);

describe('Splash describe', () => {
  test('render snap', () => {
    const container = render(component);
    expect(container.toJSON()).toMatchSnapshot();
  });
  it('wrapp', () => {
    const {getByTestId} = render(component);
    const container = getByTestId('SplashWrapp');
    expect(container).toBeDefined();
  });
});
