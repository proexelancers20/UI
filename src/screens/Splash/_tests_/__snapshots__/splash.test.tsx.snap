// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash describe render snap 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "NunitoSans_10pt-Bold",
        "fontSize": 85,
      }
    }
    testID="SplashWrapp"
  >
    Wrapp
  </Text>
</View>
`;
